src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php

```diff
@@ -42,15 +42,17 @@ class DumpDataCollector extends DataCollector implements DataDumperInterface
     private ?RequestStack $requestStack;
     private DataDumperInterface|Connection|null $dumper;
     private mixed $sourceContextProvider;
+    private bool $webMode;
 
-    public function __construct(Stopwatch $stopwatch = null, string|FileLinkFormatter $fileLinkFormat = null, string $charset = null, RequestStack $requestStack = null, DataDumperInterface|Connection $dumper = null)
+    public function __construct(Stopwatch $stopwatch = null, string|FileLinkFormatter $fileLinkFormat = null, string $charset = null, RequestStack $requestStack = null, DataDumperInterface|Connection $dumper = null, bool $webMode = null)
     {
         $fileLinkFormat = $fileLinkFormat ?: \ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format');
         $this->stopwatch = $stopwatch;
         $this->fileLinkFormat = $fileLinkFormat instanceof FileLinkFormatter && false === $fileLinkFormat->format('', 0) ? false : $fileLinkFormat;
         $this->charset = $charset ?: \ini_get('php.output_encoding') ?: \ini_get('default_charset') ?: 'UTF-8';
         $this->requestStack = $requestStack;
         $this->dumper = $dumper;
+        $this->webMode = $webMode ?? !\in_array(\PHP_SAPI, ['cli', 'phpdbg'], true);
```

Comment:
This weird to have `web` in the name while it can be CLI? Maybe `phpMode` or `runtimeType`?

Reply:
This is a boolean: true in web mode, false in cli mode.